modules/jvm/api/v9/module.yaml

```diff
@@ -0,0 +1,59 @@
+schema_version: 1
+name: jboss.container.java.jvm.api
+version: '9.0'
+description: ^
+  Defines environment variables used for configuring Java JVM, e.g. GC settings,
+  etc.  Modules providing support for configuring JVM settings should use the
+  variables listed within this module to configure the JVM appropriately.
+
+envs:
+- name: JAVA_OPTS
+  description: JVM options passed to the `java` command.
+  example: "-verbose:class"
+- name: JAVA_OPTS_APPEND
+  description: User specified Java options to be appended to generated options in JAVA_OPTS.
+  example: "-Dsome.property=foo"
+- name: JAVA_MAX_MEM_RATIO
+  description: Is used when no `-Xmx` option is given in **JAVA_OPTS**. This is used to calculate a default maximal heap memory based on a containers restriction. If used in a container without any memory constraints for the container then this option has no effect. If there is a memory constraint then `-Xmx` is set to a ratio of the container available memory as set here. The default is `50` which means 50% of the available memory is used as an upper boundary. You can skip this mechanism by setting this value to `0` in which case no `-Xmx` option is added.
+- name: JAVA_INITIAL_MEM_RATIO
+  description: Is used when no `-Xms` option is given in **JAVA_OPTS**. This is used to calculate a default initial heap memory based on the maximum heap memory. If used in a container without any memory constraints for the container then this option has no effect. If there is a memory constraint then `-Xms` is set to a ratio of the `-Xmx` memory as set here. The default is `25` which means 25% of the `-Xmx` is used as the initial heap size. You can skip this mechanism by setting this value to `0` in which case no `-Xms` option is added.
+  example: "25"
+- name: JAVA_MAX_INITIAL_MEM
+  description: Is used when no `-Xms` option is given in **JAVA_OPTS**. This is used to calculate the maximum value of the initial heap memory. If used in a container without any memory constraints for the container then this option has no effect. If there is a memory constraint then `-Xms` is limited to the value set here. The default is 4096MB which means the calculated value of `-Xms` never will be greater than 4096MB. The value of this variable is expressed in MB.
+  example: "4096"
+- name: JAVA_DIAGNOSTICS
+  description: Set this to get some diagnostics information to standard output when things are happening. **Note:** This option, if set to true, will set `-XX:+UnlockDiagnosticVMOptions`. **Disabled by default.**
+  example: "true"
+- name: JAVA_DEBUG
+  description: If set remote debugging will be switched on. **Disabled by default.**
+  example: "true"
+- name: JAVA_DEBUG_PORT
+  description: Port used for remote debugging. Defaults to *5005*.
+  example: "8787"
+- name: CONTAINER_CORE_LIMIT
+  description: A calculated core limit as described in https://www.kernel.org/doc/Documentation/scheduler/sched-bwc.txt.
+  example: "2"
+- name: CONTAINER_MAX_MEMORY
```

Comment:
Not sure what this does. Is this setting `-XX:MaxRAM`?

Reply:
It's not intended to be set by the user, it's set by the scripts which inspect the cgroups *v1* filesystem layouts for memory limits. It influences the setting of -Xmx I think, and possibly others. It should probably go.

Reply:
Yes, it looks like it's set in `container_memory` and used by `max_memory` and friends. The latter we should be able to get rid of and use the JVM's `*Percentage` flags instead.